bundles/specmate-jira-connector/src/com/specmate/connectors/jira/internal/services/JiraConnector.java

```diff
@@ -0,0 +1,223 @@
+package com.specmate.connectors.jira.internal.services;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.log.LogService;
+
+import com.atlassian.jira.rest.client.api.JiraRestClient;
+import com.atlassian.jira.rest.client.api.RestClientException;
+import com.atlassian.jira.rest.client.api.domain.Issue;
+import com.atlassian.jira.rest.client.api.domain.SearchResult;
+import com.atlassian.jira.rest.client.internal.async.AsynchronousJiraRestClientFactory;
+import com.specmate.common.SpecmateException;
+import com.specmate.common.SpecmateValidationException;
+import com.specmate.connectors.api.IRequirementsSource;
+import com.specmate.connectors.config.ProjectConfigService;
+import com.specmate.connectors.jira.config.JiraConnectorConfig;
+import com.specmate.model.base.BaseFactory;
+import com.specmate.model.base.Folder;
+import com.specmate.model.base.IContainer;
+import com.specmate.model.requirements.Requirement;
+import com.specmate.model.requirements.RequirementsFactory;
+
+@Component(immediate = true, service = IRequirementsSource.class, configurationPid = JiraConnectorConfig.PID, configurationPolicy = ConfigurationPolicy.REQUIRE)
+public class JiraConnector implements IRequirementsSource {
+
+	private LogService logService;
+	private String id;
+	private JiraRestClient jiraClient;
+	private String projectName;
+	private String url;
+	
+	private Map<Issue, Folder> epicFolders = new HashMap<>();
+	private Map<Requirement, Issue> requirmentEpics = new HashMap<>();
+	private Folder defaultFolder;
+
+	@Activate
+	public void activate(Map<String, Object> properties) throws SpecmateValidationException {
+		validateConfig(properties);
+		
+		id = (String) properties.get(ProjectConfigService.KEY_CONNECTOR_ID);
+		url = (String) properties.get(JiraConnectorConfig.KEY_JIRA_URL);
+		projectName = (String) properties.get(JiraConnectorConfig.KEY_JIRA_PROJECT);
+		String username = (String) properties.get(JiraConnectorConfig.KEY_JIRA_USERNAME);
+		String password = (String) properties.get(JiraConnectorConfig.KEY_JIRA_PASSWORD);
+		
+		try {
+			jiraClient = new AsynchronousJiraRestClientFactory().createWithBasicHttpAuthentication(new URI(url), username, password);
+		} catch (URISyntaxException e) {
+			e.printStackTrace();

```

Comment:
I'd throw here a SpecmateException. Otherwise, the following code will give unexpected results and produce misleading log entries.